core/src/main/java/org/elasticsearch/cluster/routing/ShardRouting.java

```diff
@@ -567,11 +575,16 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                 .field("node", currentNodeId())
                 .field("relocating_node", relocatingNodeId())
                 .field("shard", shardId().id())
-                .field("index", shardId().index().name());
+                .field("index", shardId().index().name())
+                .field("version", version);
+
         if (restoreSource() != null) {
             builder.field("restore_source");
             restoreSource().toXContent(builder, params);
         }
+        if (allocationId != null) {
+            allocationId.toXContent(builder, params);
+        }
```

Comment:
why did you add it? I mean, it is very internal...., it will mean cluster state API will become so much more noisy